omisego-sdk/src/main/java/co/omisego/omisego/OMGAPIClient.kt

```diff
@@ -91,15 +75,14 @@ class OMGAPIClient(private val eWalletClient: EWalletClient) {
      *
      * @param callback A callback to receive the response from server.
      */
-    fun listBalances(callback: Callback<BalanceList>) {
-        val type = object : TypeToken<OMGResponse<BalanceList>>() {}.type
-        val callbackManager = CallbackManager<BalanceList>(serializer, type)
-        eWalletClient.eWalletAPI.listBalance().enqueue(callbackManager.transform(callback))
-    }
+    fun listBalances() = eWalletClient.eWalletAPI.listBalances()
 
-    private inline fun <reified T> enqueueCall(callback: Callback<T>, call: EWalletAPI.() -> Call<JsonElement>){
-        val type = object : TypeToken<OMGResponse<T>>() {}.type
-        val callbackManager = CallbackManager<T>(serializer, type)
-        eWalletClient.eWalletAPI.run(call).enqueue(callbackManager.transform(callback))
+    /**
+     * Set new [authenticationToken].
+     *
+     * @param authenticationToken An authentication token to replace the old value.
+     */
+    fun setAuthenticationToken(authenticationToken: String) {
```

Comment:
The reason I'm adding this method is when the client calls `logout()`, then it should be setting new `authenticationToken` value because the old token is already expired. If we don't have this method, they should re-initialize the `EWalletClient`. I don't sure that the developer wants to re-initialize the `EWalletClient` every time the `logout()` is called. 🤔

Reply:
I just renamed it to `setAuthenticationTokenHeader` to make it clearer